EtherealEngine/etherealengine
# 7461 Optimize avatar load

## Summary

A lot of refactor work was done by @AidanCaruso avatar-fx branch. I've ensured when we are updating the uniforms for the animation that we don't traverse the entire model each time. I have fixed a race condition where if your avatar leaves quickly after joining the effect will remain. As well as some small undefined bugs here and there.

## References

#7461 


## Checklist
- [ ] [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- [ ] Changes reviewed by at least 2 approved reviewer


## QA Steps

I want to see how this performs now on VR, Phone, and Laggy macbook. If not well. I will disable for those platforms.
